xcdat/temporal.py

```diff
@@ -695,15 +741,18 @@ def _temporal_avg(
         ):
             ds = self._drop_incomplete_djf(ds)
 
-        # Group the time coordinates and average the data variable using them.
-        self._time_grouped = self._group_time_coords(ds.cf["T"])
         dv = _get_data_var(ds, data_var)
-        dv = self._averager(dv)
 
-        # The dataset's original "time" dimension becomes obsolete after
-        # calculating the climatology of the data variable, so it is dropped
-        # and replaced.
-        ds = ds.drop_dims("time")
+        if self._mode == "average":
+            dv = self._average(dv)
+        else:
+            dv = self._group_average(dv)
+
+            # The original time dimension becomes obsolete after time averaging
+            # the data variable so it is dropped. Adding the time averaged
```

Comment:
```suggestion
            # The original time dimension is dropped from the Dataset because 
            # it becomes  after the data variable is averaged. A new time dimension
            # will be added to the Dataset when adding the averaged data variable.
```